docs/site/tutorials/todo-list/todo-list-tutorial-has-one-relation.md

```diff
@@ -0,0 +1,262 @@
+```
+
+### Create the Repository
+
+Using `lb4 repository`, let's create the repository:
+
+```sh
+lb4 repository
+? Please select the datasource DbDatasource
+? Select the model(s) you want to generate a repository TodoListImage
+? Please select the repository base class DefaultCrudRepository (Legacy juggler bridge)
+   create src/repositories/todo-list-image.repository.ts
+   update src/repositories/index.ts
+
+Repository TodoListImageRepository was created in src/repositories/
+```
+
+### Add the Relation
+
+{% include note.html content="
+We are working on adding `hasOne` to the CLI command `lb4 relation`. See [issue #2980](https://github.com/strongloop/loopback-next/issues/2980).
+" %}
+
+Adding a [`hasOne` relation](../../hasOne-relation.md) is simple. First, let's
+add the relation to the model classes:
+
+{% include code-caption.html content="src/models/todo-list-image.model.ts" %}
+
+```ts
+import {belongsTo} from '@loopback/repository';
+import {TodoList, TodoListWithRelations} from './todo-list.model';
+
+@model()
+export class TodoListImage extends Entity {
+  // ... other properties
+
+  @belongsTo(() => TodoList)
+  todoListId: number;
+
+  // ...
+}
+
+export interface TodoListImageRelations {
+  todoList?: TodoListWithRelations;
+}
+```
+
+{% include code-caption.html content="src/models/todo-list.model.ts" %}
+
+```ts
+import {hasOne} from '@loopback/repository';
+import {
+  TodoListImage,
+  TodoListImageWithRelations,
+} from './todo-list-image.model';
+
+@model()
+export class TodoList extends Entity {
+  // ... other properties
+
+  @hasOne(() => TodoListImage)
+  image?: TodoListImage;
+
+  // ...
+}
+
+export interface TodoListRelations {
+  todos?: TodoWithRelations[];
+
+  // Add the following line
+  image?: TodoListImageWithRelations;
+}
+```
+
+Next, let's add the relation to the repository classes:
+
+{% include code-caption.html content="src/repositories/todo-list.repository.ts" %}
+
+```ts
+// Add the following imports
+import {HasOneRepositoryFactory} from '@loopback/repository';
+import {TodoListImage} from '../models';
+import {TodoListImageRepository} from './todo-list-image.repository';
+
+export class TodoListRepository extends DefaultCrudRepository<
+  TodoList,
+  typeof TodoList.prototype.id,
+  TodoListRelations
+> {
+  // other code
+
+  // Add the following
+  public readonly image: HasOneRepositoryFactory<
+    TodoListImage,
+    typeof TodoList.prototype.id
+  >;
+
+  constructor(
+    // other code
+
+    // Add the following
+    @repository.getter('TodoListImageRepository')
+    protected todoListImageRepositoryGetter: Getter<TodoListImageRepository>,
+  ) {
+    // other code
+
+    // Add the following
+    this.image = this.createHasOneRepositoryFactoryFor(
+      'image',
+      todoListImageRepositoryGetter,
+    );
+
+    this.registerInclusionResolver('image', this.image.inclusionResolver);
+  }
+}
+```
+
+```ts
+import {BelongsToAccessor} from '@loopback/repository';
```

Comment:
Hmm, a question: since we are in the hasOne relation's document, is the `TodoListImageRepository` with belongsTo needed?

Maybe add a few words to explain why a belongsTo relation's setup is needed for a hasOne relation :)

Reply:
> Maybe add a few words to explain why a belongsTo relation's setup is needed for a hasOne relation

I don't think belongsTo relation is needed for hasOne / hasMany. They are independent. 
Personally I'd prefer to just have hasOne setup here. Then explain that the corresponding belongsTo can be added as the same way we did for hasMany+belongsTo if needed.

Either way is fine. I just don't want users to have a wrong impression that belongsTo has to be existed with hasOne or hasMany.